### DISCUSSION TITLE
Incorrect Tiff files thumbnail

### DISCUSSION THREAD

**Discussion by amanraj8650** [ImageMagick/ImageMagick#7480] [General]
I downloaded two thumbnails of same tif file using ImageMagick command (800 pixel and 3200 pixel).
The 800 pixel thumbnail seems correct and its color is similar to original file, but 3200 pixel thumbnails seems incorrect and its color looks very different.
Attaching sample thumbnails of file-
800 px thumbnail:
<img width="322" alt="image" src="https://github.com/user-attachments/assets/95e57609-e361-46bd-b2e2-a6f1bfa6c995">

3200 px thumbnail:
<img width="316" alt="image" src="https://github.com/user-attachments/assets/6866f213-cb79-45e0-ac34-60f5bf47b170">

Why this is happening and how it can be fixed?

**Comment by snibgo** [comment]
I can't guess.

Please show the commands you used, and the input image.

**Reply by amanraj8650** [reply]
Compressed input file: [Servitec_75_Touch-Canto.zip](https://github.com/user-attachments/files/16243799/Servitec_75_Touch-Canto.zip)
The file size is big so i compressed it.

Commands:
"convert +repage " + args[0] + " " + args[1] + "[0] " + args[2]"

e.g.
For 800 px: 
convert +repage    -resize 800x800  /usr/local/tomcat/temp/56a4ee6db876466d93ad755c56981c09-ac075e9c28944cc89ebe6700e5ac988f-convOutput.jpg[0] /usr/local/tomcat/temp/56a4ee6db876466d93ad755c56981c09-ac075e9c28944cc89ebe6700e5ac988f-convOutput.jpg

For 3200 px: 
convert +repage    -resize 3200x3200  /usr/local/tomcat/temp/56a4ee6db876466d93ad755c56981c09-ac075e9c28944cc89ebe6700e5ac988f-convOutput.jpg[0] /usr/local/tomcat/temp/56a4ee6db876466d93ad755c56981c09-ac075e9c28944cc89ebe6700e5ac988f-convOutput.jpg

**Comment by snibgo** [comment]
I can't read the zip file. It seems to be corrupt:

> Archive:  Servitec_75_Touch-Canto.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.

The images you showed are:
```
C:\web\im>%IMG7%magick greenthing*.png info:
greenthing1.png[0] PNG 643x874 643x874+0+0 8-bit TrueColorAlpha sRGB 123715B 0.001u 0:00.000
greenthing2.png[1] PNG 631x865 631x865+0+0 8-bit TrueColorAlpha sRGB 190255B 0.003u 0:00.002
```
Note that no dimension is 800, and no dimension is 3200. So those commands did not make those files.

What version of IM are you using? If v7, I suggest using `magick` instead of `convert`. Also put the options in the order they are to be executed: read the file, resize and repage, write the file. This shouldn't affect the colours.

**Reply by amanraj8650** [reply]
Can you rename the .zip to .7z and try extracting file. Ideally I can't upload file more than 25 MB here, so I compressed it to .7z and renamed to .zip as here .7z file type is not supported.
And below are files:
800px:
![56a4ee6db876466d93ad755c56981c09 800](https://github.com/user-attachments/assets/71805c53-e79b-4d99-9c72-4f6dea591465)

3200px:
![56a4ee6db876466d93ad755c56981c09 3200](https://github.com/user-attachments/assets/21062707-1070-45d3-be23-ad60da4d1dee)

**Comment by snibgo** [comment]
> Can you rename the .zip to .7z and try extracting file.

That fixes the problem.

```
magick Servitec_75_Touch-Canto.tif info:

Servitec_75_Touch-Canto.tif[0] TIFF 2480x3507 2480x3507+0+0 16-bit ColorSeparation CMYK 40.2456MiB 0.000u 0:00.000
Servitec_75_Touch-Canto.tif[1] TIFF 1151x2168 1151x2168+688+758 16-bit CMYK 40.2456MiB 0.001u 0:00.001
```
So the file contains two images, both CMYK. `-verbose info:` tells me that neither has an ICC color profile, so we don't know how to correctly convert the images to sRGB.

The two images you show in your latest post are:
```
magick greent?.jpg info:
greent1.jpg[0] JPEG 566x800 566x800+0+0 8-bit sRGB 30306B 0.000u 0:00.000
greent2.jpg[1] JPEG 2263x3200 2263x3200+0+0 8-bit CMYK 649657B 0.004u 0:00.004
```
So the dimensions are consistent with `-resize 800x800` and `-resize 3200x3200`.

However, the posted file `Servitec_75_Touch-Canto.tif` is a TIFF, but your commands have input files with extension `.jpg`. So something you have posted is not correct.

**Reply by amanraj8650** [reply]
Yes, you are right the original file is TIFF format. But during upload process in our application we are generating previews/thumbnails of this file which is of type JPG using imagemagick convert command.
If you see the previews attached you can see the color difference (one is light green, other is dark shade of green)

**Comment by snibgo** [comment]
Resizing an image to two different sizes won't change the colours in the way you show.

So I suppose you are doing different processing for the two sizes.

**Reply by amanraj8650** [reply]
Processing is done using same commands, how it is happening on ImageMagick side that is a concern.
And we have tried for multiple TIF files, and it generates correct thumbnail for them of different sized. This issue is observed for this specific file.

Can you try it on your end and see if you also get similar images that i shared to you above.

**Comment by snibgo** [comment]
I have tried, and I get the expected results. I cannot reproduce your problem.

You still have not shown your commands that read the tiff file and makes the two outputs.

**Comment by amanraj8650** [comment]
For 800 px:
convert +repage -resize 800x800 /usr/local/tomcat/temp/56a4ee6db876466d93ad755c56981c09-ac075e9c28944cc89ebe6700e5ac988f-convOutput.jpg[0] /usr/local/tomcat/temp/56a4ee6db876466d93ad755c56981c09-ac075e9c28944cc89ebe6700e5ac988f-convOutput.jpg

For 3200 px:
convert +repage -resize 3200x3200 /usr/local/tomcat/temp/56a4ee6db876466d93ad755c56981c09-ac075e9c28944cc89ebe6700e5ac988f-convOutput.jpg[0] /usr/local/tomcat/temp/56a4ee6db876466d93ad755c56981c09-ac075e9c28944cc89ebe6700e5ac988f-convOutput.jpg

These are the only commands, we read the TIF files and many other formats we have we write down into a file and append JPG as extension as these are preview/thumbnails that are to be shown on UI page and then use above ImageMagick commands.

I also checked both preview looks similar when i open with some image software, but on UI color difference is there. Not sure how its happening.

**Comment by snibgo** [comment]
As I have already said, those commands don't read a TIFF file.

**Reply by fmw42** [reply]
What version of Imagemagick are you using?  If IM 7, then use magick in place of convert and reorder your commands to read the input image first as: 

`magick image settings operators output`

Probably wise also for IM 6.

**Reply by amanraj8650** [reply]
I have used ExifTool to check metadata of both above previews and found that-
800px have attribute **Color Space Data : RGB**
3200px don't have this attribute.

Do you think imagemagick append this metadata during processing? As this can be issue resulting in color difference
